source/modulo_components/modulo_components/component_interface.py

```diff
@@ -460,25 +460,31 @@ def add_input(self, signal_name: str, subscription: Union[str, Callable], messag
                 self._inputs[parsed_signal_name] = self.create_subscription(message_type, topic_name, subscription,
                                                                             self._qos)
             elif isinstance(subscription, str):
-                if user_callback:
+                if callable(user_callback):
                     signature = inspect.signature(user_callback)
                     if len(signature.parameters) != 0:
                         raise AddSignalError("Provide a user callback that has no input arguments.")
+                else:
+                    if user_callback:
+                        self.get_logger().warn("Provided user callback is not a callable, ignoring it.")
+                    user_callback = lambda: None
```

Comment:
It would be good to avoid PEP warnings. Just define the function object at the lowest scope 
```suggestion
                    def default_callback():
                        return None
                    user_callback = default_callback
```